File: Java/src/sorts/BubbleSort.java
package sorts;

import util.SortUtil;

/**
 * Created by Chris Keyser on 3/17/2016.
 * Bubble sort implementation
 */
public class BubbleSort {
    public static int[] sort(int[] arr) {
        while(!SortUtil.isSorted(arr)) {
            for(int i = 0; i < arr.length - 1; i++) {
                if(arr[i] > arr[i + 1]) {
                    SortUtil.swap(arr, i, i + 1);
                }
            }
        }

        return arr;
    }
}

File: js/src/sorts/bubbleSort.js
function BubbleSort() {}

// standard BubbleSort implementation
// has O(n^2) time complexity
// while very simple, is also never useful in the real world
// due to its extreme inefficiency
BubbleSort.prototype = {
  sort: function(arr) {
    var input = arr.slice(0);

    // until we're done sorting
    while(!util.isSorted(input)) {
      // check each index value to see if it's greater than index + 1 value
      // if it is, swap them. Repeat until sorted
      for(var i = 0; i < input.length - 1; i++) {
        if(input[i] > input[i + 1]) {
          util.swap(input, i, i + 1);
        }
      }
    }

    return input;
  }
}
